semana4/lista-de-exercicios/index.js

```diff
@@ -0,0 +1,419 @@
+o segundo imprime o o valor de numero2 que será o maior número do array numeros
+
+=======================================================================================================
+
+EXERCÍCIOS DE LÓGICA DE PROGRAMAÇÃO  (como proposto estou colocando os exercícios dentro de funções não sei bem se da forma como foi solicitado, mas foi como eu entendi para fazer)
+
+EXERCÍCIO 1
+
+Usando as estruturas de laços como o while, for-of ou até mesmo uma função callback como o forEach
+
+*/
+
+function iteraLista(listaDeNumeros){ 
+    
+    let dobroListaDeNumeros = []
+    let listaDeNegativos = []
+    
+    
+    let i = 0
+    while (i < listaDeNumeros.length) {   // exemplo de iteração com while
+        let novoNumero = listaDeNumeros[i]*2
+        dobroListaDeNumeros.push(novoNumero)
+        i++
+    }
+    
+    
+    for(numero of listaDeNumeros) {     // exemplo de iteraçao com for-of        
+        let numeroNegativo = numero*-1
+        listaDeNegativos.push(numeroNegativo)
+    }
+     
+    let numerosPares = []
+    listaDeNumeros.forEach((numero, index, array) =>{    // exemplo de iteração com forEach
+        if (numero % 2 === 0) {
+            numerosPares.push(numero)
+        }
+    })
+    
+    console.log(dobroListaDeNumeros)
+    console.log(listaDeNegativos)
+    console.log(numerosPares)
+
+}
+
+/*
+
+EXERCÍCIO 2 
+
+a) false
+
+b) false
+
+c) true
+
+d) true
+
+e) true
+
+-----------------------------------------------------------------------------------------------
+
+
+EXERCÍCIO 3
+
+O código não funcionará adequadamente primeiro porque quantidadeDeNumeros está declarado como const mas sem valor,
+e isso ira retornar um erro no console, outro problema é que se quisermos que esse número seja digitado pelo usuário
+precisáriamos atribuir ela a um prompt e quantidadeDeNumeros sendo declarado com const não vai nos permitir alterar seu valor
+Um outro erro no código que notei é que a condição de parada do while está como i maior ou igual ao número passado, isso irá imprimir um número a mais do que o esperado
+E por ultimo a variável i tem o valor 0 e está sem incremento dentro do laço, isso vai gerar um loop infinito.
+Abaixo o código com algumas alterações para conseguirmos o comportamento esperado.
+
+*/
+
+function quantidadeDePares() {  
+    let quantidadeDeNumerosPares = Number(prompt("Digite um número:"))
+    let i = 0 
+    while(i < quantidadeDeNumerosPares) {
+        console.log(i*2)
+        i++
+    }
+}
+
+//--------------------------------------------------------------------------------------
+// EXERCÍCIO 4
+
+function classificaTriangulo (a, b, c) {
+    let classificacao = ""
+    if (a === 0 || b === 0 || c === 0) {
+        classificacao = "Não é triângulo"
+    }else if (a === b && b === c) {
+        classificacao = "Equilátero"
+    } else if (a === b || a === c || c === b) {
+        classificacao = "Isósceles"    
+    } else {
```

Comment:
Na verdade ele vai ser "Escaleno"

Reply:
Nossa, verdade, foi falta de atenção essa, e o exercício todo, fui copiar um trecho do código acho e não me atentei em mudar as condições.

Reply:
nossa verdade, estava revisando esse exercício e ele ficou errado, quis introduzir mais uma condição que seria caso um lado fosse = 0 e acabei não me atentando em mudar as outras condições corretamente.

Reply:
para ser escaleno não teria que ter os três lados diferentes? O que eu tentei fazer na condição foi falar que se dois dos lados forem iguais ele será isósceles. Não é isso?

Reply:
verdade, falta de atenção ai, tinha tentado fazer uma outra lógica antes e não corrigi esse final.